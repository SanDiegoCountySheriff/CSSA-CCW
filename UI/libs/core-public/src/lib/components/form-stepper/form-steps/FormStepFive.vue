<template>
  <div>
    <v-form
      ref="form"
      v-model="state.valid"
    >
      <v-subheader class="sub-header font-weight-bold">
        {{ $t('Contact information') }}
      </v-subheader>
      <v-row>
        <v-col
          cols="6"
          md="5"
          sm="3"
        >
          <v-text-field
            :label="$t('Primary phone number')"
            :rules="[v => !!v || $t('Primary phone number cannot be blank')]"
            :v-model="state.contact.primaryPhoneNumber"
          />
        </v-col>
        <v-col
          cols="6"
          md="5"
          sm="3"
        >
          <v-text-field
            :label="$t('Cell phone number')"
            v-model="state.contact.cellPhoneNumber"
          />
        </v-col>

        <v-col
          cols="6"
          md="5"
          sm="3"
        >
          <v-text-field
            :label="$t('Work phone number')"
            v-model="state.contact.workPhoneNumber"
          />
        </v-col>

        <v-col
          cols="6"
          md="5"
          sm="3"
        >
          <v-text-field
            :label="$t('Fax number')"
            v-model="state.contact.faxPhoneNumber"
          />
        </v-col>
      </v-row>
      <v-row>
        <v-col
          cols="6"
          md="5"
          sm="3"
        >
          <CheckboxInput
            :label="'Text message updates'"
            :target="'textMessageUpdates'"
            @input="
              v => {
                state.contact.textMessageUpdates = v;
              }
            "
          />
        </v-col>
      </v-row>
    </v-form>
    <FormButtonContainer
      :valid="state.valid"
      @submit="handleSubmit"
    />
  </div>
</template>

<script setup lang="ts">
import { reactive } from 'vue';
import { useContactStore } from '@shared-ui/stores/contact';
import { ContactInfoType } from '@shared-utils/types/defaultTypes';
import CheckboxInput from '@shared-ui/components/inputs/CheckboxInput.vue';
import FormButtonContainer from '@core-public/components/containers/FormButtonContainer.vue';

interface FormStepFiveProps {
  handleNextSection: () => void;
}

const props = withDefaults(defineProps<FormStepFiveProps>(), {
  handleNextSection: () => null,
});

const state = reactive({
  contact: {
    textMessageUpdates: false,
  } as ContactInfoType,
  valid: false,
});

const contactInfoStore = useContactStore();

function handleSubmit() {
  contactInfoStore.setContactInfo(state.contact);
  props.handleNextSection();
}
</script>
