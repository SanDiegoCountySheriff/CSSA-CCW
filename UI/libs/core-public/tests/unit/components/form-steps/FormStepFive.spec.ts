import FormStepFive from '@core-public/../../../../../shared/ui/src/lib/components/form-stepper/form-steps/FormStepFive.vue';
import Vuetify from 'vuetify';
import { createLocalVue, mount } from '@vue/test-utils';
import { createTestingPinia } from '@pinia/testing';

const pinia = createTestingPinia();
const localVue = createLocalVue();
const tMock = {
  $t: text => text,
};

describe('FormStepFive', () => {
  let vuetify = new Vuetify();
  let wrapper;

  beforeEach(() => {
    vuetify = new Vuetify();
    wrapper = mount(FormStepFive, {
      localVue,
      vuetify,
      pinia,
      mocks: tMock,
      propsData: {
        handleNextSection: () => null,
      },
    });
  });
  it('Should match the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
