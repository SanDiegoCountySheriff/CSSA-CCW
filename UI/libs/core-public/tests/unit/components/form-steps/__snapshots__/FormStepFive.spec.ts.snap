// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormStepFive Should match the snapshot 1`] = `
<div handlenextsection="() => null">
  <form novalidate="novalidate" class="v-form">
    <div class="v-subheader sub-header font-weight-bold theme--light">
      Contact information
    </div>
    <div class="row ml-5">
      <div class="col-sm-3 col-md-5 col-6">
        <div class="v-input theme--light v-text-field">
          <div class="v-input__prepend-outer"><i aria-hidden="true" class="v-icon notranslate mdi mdi-asterisk theme--light error--text" style="font-size: 12px;"></i></div>
          <div class="v-input__control">
            <div class="v-input__slot">
              <div class="v-text-field__slot"><label for="input-5" class="v-label theme--light" style="left: 0px; position: absolute;">Primary phone number</label><input id="input-5" type="text"></div>
            </div>
            <div class="v-text-field__details">
              <div class="v-messages theme--light">
                <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div class="col-sm-3 col-md-5 col-6">
        <div class="v-input theme--light v-text-field">
          <div class="v-input__control">
            <div class="v-input__slot">
              <div class="v-text-field__slot"><label for="input-9" class="v-label theme--light" style="left: 0px; position: absolute;">Cell phone number</label><input id="input-9" type="text"></div>
            </div>
            <div class="v-text-field__details">
              <div class="v-messages theme--light">
                <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div class="col-sm-3 col-md-5 col-6">
        <div class="v-input theme--light v-text-field">
          <div class="v-input__control">
            <div class="v-input__slot">
              <div class="v-text-field__slot"><label for="input-12" class="v-label theme--light" style="left: 0px; position: absolute;">Work phone number</label><input id="input-12" type="text"></div>
            </div>
            <div class="v-text-field__details">
              <div class="v-messages theme--light">
                <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div class="col-sm-3 col-md-5 col-6">
        <div class="v-input theme--light v-text-field">
          <div class="v-input__control">
            <div class="v-input__slot">
              <div class="v-text-field__slot"><label for="input-15" class="v-label theme--light" style="left: 0px; position: absolute;">Fax number</label><input id="input-15" type="text"></div>
            </div>
            <div class="v-text-field__details">
              <div class="v-messages theme--light">
                <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div class="row">
      <div class="col-sm-3 col-md-5 col-6">
        <div class="v-input theme--light v-input--selection-controls v-input--checkbox">
          <div class="v-input__control">
            <div class="v-input__slot">
              <div class="v-input--selection-controls__input"><i aria-hidden="true" class="v-icon notranslate mdi mdi-checkbox-blank-outline theme--light"></i><input aria-checked="false" id="input-19" role="checkbox" type="checkbox" value="">
                <div class="v-input--selection-controls__ripple"></div>
              </div><label for="input-19" class="v-label theme--light" style="left: 0px; position: relative;">Text message updates</label>
            </div>
            <div class="v-messages theme--light">
              <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
            </div>
          </div>
        </div>
      </div>
    </div>
  </form>
  <div class="form-btn-container"><button type="button" disabled="disabled" class="v-btn v-btn--disabled v-btn--has-bg theme--light v-size--default"><span class="v-btn__content">
    Continue
  </span></button> <button type="button" class="v-btn v-btn--is-elevated v-btn--has-bg theme--light v-size--default secondary mx-2"><span class="v-btn__content">
    Save and Exit
  </span></button> <button type="button" class="v-btn v-btn--is-elevated v-btn--has-bg theme--light v-size--default error mr-2"><span class="v-btn__content">
    Cancel
  </span></button></div>
</div>
`;
